tests/scanners/test.javascript.js

```diff
@@ -141,18 +143,68 @@ describe('JavaScript Scanner', function() {
   });
 
   // Depends on: https://github.com/mozilla/addons-validator/issues/7
-  it.skip('ignores /*eslint-disable*/ comments', () => {
-    var code = '/*eslint-disable*/\n';
-    code += 'var myDatabase = indexeddb || mozIndexedDB;';
+  it('ignores /*eslint-disable*/ comments', () => {
+    var code = singleLineString`/*eslint-disable*/
+                                var myDatabase = indexeddb || mozIndexedDB;`;
+    var jsScanner = new JavaScriptScanner(code, 'badcode.js');
+
+    return jsScanner.scan()
+      .then((validationMessages) => {
+        assert.equal(validationMessages.length, 1);
+        assert.equal(validationMessages[0].code, messages.MOZINDEXEDDB.code);
+      });
+  });
+
+  it('ignores // eslint-disable-line comments', () => {
+    var code = singleLineString`var myDatabase = indexeddb || mozIndexedDB;
+                                // eslint-disable-line`;
     var jsScanner = new JavaScriptScanner(code, 'badcode.js');
 
     return jsScanner.scan()
       .then((validationMessages) => {
         assert.equal(validationMessages.length, 1);
-        assert.equal(validationMessages[0].id, 'mozIndexedDB');
+        assert.equal(validationMessages[0].code, messages.MOZINDEXEDDB.code);
+      });
+  });
+
+  // This should not cause a syntax error; it should still be parsing code
+  // as ES6 because it ignores the env change.
+  it('ignores /*eslint-env*/ comments', () => {
```

Comment:
Looks like it never worked, I’ll add a test that _adds_ an env then, which seems the only option.